runhouse/resources/hardware/cluster.py

```diff
@@ -1957,3 +1964,195 @@ def _folder_mv(
+        if sky_live_clusters:
+            return [
+                cluster
+                for cluster in sky_live_clusters
+                if f'/{rns_client.username}/{cluster.get("name")}'
+                not in den_clusters_names
+            ]
+        else:
+            return []
+
+    @staticmethod
+    def _get_running_and_not_running_clusters(clusters: list):
+        running_clusters, not_running_clusters = [], []
+
+        for den_cluster in clusters:
+            # get just name, not full rns address. reset is used so the name will be printed all in white.
+            cluster_name = f'[reset]{den_cluster.get("name").split("/")[-1]}'
+            cluster_type = den_cluster.get("data").get("resource_subtype")
+            cluster_status = (
+                den_cluster.get("status") if den_cluster.get("status") else "unknown"
+            )
+
+            # currently relying on status pings to den as a sign of cluster activity.
+            # The split is required to remove milliseconds and the offset (according to UTC) from the timestamp.
+            # (status_last_checked is in the following format: YYYY-MM-DD HH:MM:SS.ssssss±HH:MM)
+
+            last_active_at = den_cluster.get("status_last_checked")
+            last_active_at = (
+                datetime.datetime.fromisoformat(last_active_at.split(".")[0])
+                if isinstance(last_active_at, str)
+                else datetime.datetime(1970, 1, 1)
+            )  # Convert to datetime
+            last_active_at = last_active_at.replace(tzinfo=datetime.timezone.utc)
+
+            if cluster_status == "running" and not last_active_at:
+                # For BC, in case there are clusters that were saved and created before we introduced sending cluster status to den.
+                cluster_status = "unknown"
+
+            cluster_info = {
+                "Name": cluster_name,
+                "Cluster Type": cluster_type,
+                "Status": cluster_status,
+                "Last Active (UTC)": last_active_at,
+            }
+            running_clusters.append(
+                cluster_info
+            ) if cluster_status == "running" else not_running_clusters.append(
+                cluster_info
+            )
+
+        # TODO: will be used if we'll need to print not-running clusters
+
+        # Sort clusters by the 'Last Active (UTC)' and 'Status' column
+        not_running_clusters = sorted(
+            not_running_clusters,
+            key=lambda x: (x["Last Active (UTC)"], x["Status"]),
+            reverse=True,
+        )
+        running_clusters = sorted(
+            running_clusters, key=lambda x: x["Last Active (UTC)"], reverse=True
+        )
+
+        return running_clusters, not_running_clusters
+
+    ###############################
+    # Cluster list
+    ###############################
+    @staticmethod
+    def list(
+        get_all_clusters: Optional[bool] = False,
```

Comment:
do we need a param for this? or can we just say if the others aren't specifed then we just bring all of them

Reply:
hmm, i actually think that maybe it is worth keeping it.

If no filters are specified, the current default behaviour is that we display only running clusters that where active in the last 24 hours. Therefore, we won't need to get all clusters (running, terminated etc), and then filter the den response to match the default behaviour. Instead, we'll just send a query to den to get the running clusters in the last 24 hours.

Reply:
rename `get_all_clusters` to `show_all`